scripts/usb_reset.py

```diff
@@ -90,6 +89,48 @@ def get_cg_dir(domid):
     return "/sys/fs/cgroup/devices/qemu-{}".format(domid)
 
 
+def get_ids_path(device):
+    usb_dir = "/var/run/nonpersistent/usb"
+    if not os.path.isdir(usb_dir):
+        os.makedirs(usb_dir)
+    return usb_dir + "/{}".format(device)
+
+
+def save_device_ids(device):
+    path = dev_path(device)
+
+    try:
+        stat = os.stat(path)
+        ids_info = "{} {}".format(stat.st_uid, stat.st_gid)
+    except OSError as e:
+        log.error("Failed to get stat of file {}: {}".format(path, str(e)))
+        exit(1)
+
+    try:
+        with open(get_ids_path(device), "w") as f:
+            f.write(ids_info)
+    except IOError as e:
+        log.error("Failed to save of file ids {}: {}".format(path, str(e)))
+        exit(1)
+
+
+def restore_device_ids(device):
+    ids_path = get_ids_path(device)
+    try:
+        with open(ids_path) as f:
+            uid, gid = map(int, f.readline().split())
```

Comment:
You can just use `read()` instead of `readline()`... since you don't actually write a newline previously.

Reply:
this could be treated as the last line(EOF), according to the document it seems to be OK:
f.readline() reads a single line from the file; a newline character (\n) is left at the end of the string, and is only omitted on the last line of the file if the file doesn’t end in a newline.

Reply:
Sure, it works but read() requires less processing and is shorter. It doesn't really matter either way.